From 5c0bfaaa64091309496f71e1af2224116c301289 Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 17 Feb 2015 15:56:15 +0000
Subject: [PATCH 29/51] Do not add a trailing slash when converting / to a
 Win32 path

This resembles MSys1' behavior on which Git's t0060 relies.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/path.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index c767db4..c583119 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -732,11 +732,11 @@ path_conv::check (const char *src, unsigned opt,
 	      need_directory = 1;
 	      *--tail = '\0';
 	    }
-	  /* Special case for "/" must set need_directory, without removing
+	  /* Special case for "/" must not set need_directory, neither remove
 	     trailing slash */
 	  else if (tail == path_copy + 1 && isslash (tail[-1]))
 	    {
-	      need_directory = 1;
+	      need_directory = 0;
 	    }
 	  path_end = tail;
 
-- 
2.10.2

